tests/kafkatest/services/connect.py

```diff
@@ -76,6 +78,41 @@ def set_configs(self, config_template_func, connector_config_templates=None):
         self.config_template_func = config_template_func
         self.connector_config_templates = connector_config_templates
 
+    def listening(self, node, port=8083):
+        try:
+            cmd = "nc -z %s %s" % (node.account.hostname, port)
+            node.account.ssh_output(cmd, allow_fail=False)
+            self.logger.debug("Connect worker started accepting connections at: '%s:%s')", node.account.hostname, port)
+            return True
+        except (RemoteCommandError, ValueError) as e:
+            return False
+
+    # Currently the Connect worker supports waiting on three modes:
+    # INSTANT: return immediately
+    # LOAD: return after discovering and loading plugins
+    # LISTEN: return after opening the REST port.
+    def start_async(self, mode='LISTEN'):
```

Comment:
i assume this is used instead of `start(self, mode='LISTEN')` due to naming conflicts? the name's a bit confusing since in fact both `LOAD` and `LISTEN` are synchronous, at least to some degree. i'm not sure just overriding with that extra parameter is even an issue. Would

```
def start(self, mode='LISTEN'):
    self.startup_mode = mode
    super(ConnectServiceBase, self).start()
```

not work?

Reply:
The call to the parent method was something that was escaping me ... Thanks!